src/ScottPlot4/ScottPlot/Plottable/VectorFieldList.cs

```diff
@@ -0,0 +1,87 @@
+﻿using ScottPlot.Drawing;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+
+namespace ScottPlot.Plottable
+{
+    /// <summary>
+    /// The VectorField displays arrows representing a 2D array of 2D vectors
+    /// </summary>
+    public class VectorFieldList : IPlottable
+    {
+        public string Label { get; set; } = null;
+        public bool IsVisible { get; set; } = true;
+        public int XAxisIndex { get; set; } = 0;
+        public int YAxisIndex { get; set; } = 0;
+
+        /// <summary>
+        /// Tuples define location and direction of vectors to display as arrows.
+        /// Users may manipulate this List to add/remove their own vectors.
+        /// </summary>
+        public readonly List<(Coordinate coordinate, CoordinateVector vector)> RootedVectors = new();
+
+        /// <summary>
+        /// Advanced configuration options that control how vectors are drawn as arrows
+        /// </summary>
+        public readonly Renderable.ArrowStyle ArrowStyle = new();
+
+        /// <summary>
+        /// Color to draw the arrows (if <see cref="Colormap"/> is null)
+        /// </summary>
+        public Color Color { get; set; } = Color.Blue;
+
+        /// <summary>
+        /// If defined, this colormap is used to color each arrow based on its magnitude
+        /// </summary>
+        public Colormap Colormap { get; set; } = null;
+
+        /// <summary>
+        /// If <see cref="Colormap"/> is defined, each arrow's magnitude 
+        /// is run through this function to get the fraction (from 0 to 1) 
+        /// along the colormap to sample from.
+        /// </summary>
+        public Func<double, double> ColormapScaler = (magnitude) => magnitude;
```

Comment:
This seems niche, this is one of the only feature requests we've ever had for vector fields after all.

Reply:
It makes sense if your vectors are 100s of units in size, but colormaps except a number from 0-1 😅